> [!NOTE]
> **Design mock — not a live Dr.CI run.** This is a hand-written preview of the proposed "Option C" look from this PR (task #132 part 3). The pills here are **shields.io stand-ins** for illustration; in production the same flat-style SVG is served by the new `/api/drci/advisorBadge` endpoint (so it can flip `analyzing → verdict` server-side without rewriting the comment). Job names / HUD links below are illustrative.

## :robot: Dr.CI — proposed advisor verdict look

<details open><summary><b>NEW FAILURES</b> - The following jobs have failed:</summary>
<p>

* [linux-jammy-cuda12.1-py3.10-gcc9 / test (default, 2, 5)](https://hud.pytorch.org/pr/pytorch/pytorch/123456#41234567890) ([gh](https://github.com/pytorch/pytorch/actions/runs/123/job/456))
    `RuntimeError: CUDA error: device-side assert triggered`
  <details><summary>AI verdict: <a href="https://hud.pytorch.org/pr/pytorch/pytorch/123456#41234567890"><img src="https://img.shields.io/static/v1?label=&style=flat&message=related&color=d1242f"></a></summary><blockquote>

  The assert fires inside the CUDA kernel launch path this PR modifies; it reproduces on the PR head and not on the merge base, so it is attributable to this change. (high confidence)

  <a href="https://hud.pytorch.org/pr/pytorch/pytorch/123456#41234567890">Full reasoning on HUD &rarr;</a>
  </blockquote></details>

* [linux-docs / build-docs (python)](https://hud.pytorch.org/pr/pytorch/pytorch/123456#41234599999) ([gh](https://github.com/pytorch/pytorch/actions/runs/123/job/789))
    `Sphinx error: Undefined label: 'torch.compile'`
  <details><summary>AI verdict: <a href="https://hud.pytorch.org/pr/pytorch/pytorch/123456#41234599999"><img src="https://img.shields.io/static/v1?label=&style=flat&message=not%20related&color=2da44e"></a></summary><blockquote>

  Pre-existing broken cross-reference in the docs tree, present on trunk; unrelated to this PR's runtime changes. (high confidence)

  <a href="https://hud.pytorch.org/pr/pytorch/pytorch/123456#41234599999">Full reasoning on HUD &rarr;</a>
  </blockquote></details>

* [linux-jammy-py3.9-gcc11 / test (distributed, 1, 3)](https://hud.pytorch.org/pr/pytorch/pytorch/123456#41234511111) ([gh](https://github.com/pytorch/pytorch/actions/runs/123/job/111))
    `torch.distributed.DistBackendError: NCCL timeout`
  <details><summary>AI verdict: <a href="https://hud.pytorch.org/pr/pytorch/pytorch/123456#41234511111"><img src="https://img.shields.io/static/v1?label=&style=flat&message=probably%20not%20related&color=94c11f"></a></summary><blockquote>

  Matches a known flaky NCCL-timeout pattern seen across unrelated PRs this week; weakly correlated with this change. (medium confidence)

  <a href="https://hud.pytorch.org/pr/pytorch/pytorch/123456#41234511111">Full reasoning on HUD &rarr;</a>
  </blockquote></details>
</p></details>

<details open><summary><b>UNCLASSIFIED FAILURES</b> - Likely due to your changes:</summary>
<p>

* [win-vs2022-cpu-py3 / test (default, 1, 2)](https://hud.pytorch.org/pr/pytorch/pytorch/123456#41234522222) ([gh](https://github.com/pytorch/pytorch/actions/runs/123/job/222))
    `LINK : fatal error LNK1104: cannot open file 'c10.lib'`
    AI verdict: <a href="https://hud.pytorch.org/pr/pytorch/pytorch/123456#41234522222"><img src="https://img.shields.io/static/v1?label=&style=flat&message=analyzing&color=9f9f9f"></a>
</p></details>

---

<details><summary><b>The full color/confidence gradient</b> (click to expand)</summary>
<p>

| state | pill |
|---|---|
| related — high | <img src="https://img.shields.io/static/v1?label=&style=flat&message=related&color=d1242f"> |
| probably related — med | <img src="https://img.shields.io/static/v1?label=&style=flat&message=probably%20related&color=e8702a"> |
| related (uncertain) — low | <img src="https://img.shields.io/static/v1?label=&style=flat&message=related%20(uncertain)&color=e0a82e"> |
| not related (uncertain) — low | <img src="https://img.shields.io/static/v1?label=&style=flat&message=not%20related%20(uncertain)&color=c9b81a"> |
| probably not related — med | <img src="https://img.shields.io/static/v1?label=&style=flat&message=probably%20not%20related&color=94c11f"> |
| not related — high | <img src="https://img.shields.io/static/v1?label=&style=flat&message=not%20related&color=2da44e"> |
| garbage | <img src="https://img.shields.io/static/v1?label=&style=flat&message=garbage&color=6e7781"> |
| inconclusive (unsure/unknown) | <img src="https://img.shields.io/static/v1?label=&style=flat&message=inconclusive&color=8b949e"> |
| analyzing (dispatched, no verdict) | <img src="https://img.shields.io/static/v1?label=&style=flat&message=analyzing&color=9f9f9f"> |

</p></details>

<details><summary><b>Actual production SVG</b> (what <code>renderBadgeSvg</code> emits for <code>related</code>, high) — click to expand</summary>

```xml
<svg xmlns="http://www.w3.org/2000/svg" width="62" height="20" role="img" aria-label="related">
<title>related</title>
<rect width="62" height="20" rx="3" fill="#d1242f"/>
<text x="31.0" y="14" fill="#ffffff" font-family="Verdana,DejaVu Sans,Geneva,sans-serif" font-size="11" text-anchor="middle">related</text>
</svg>
```

</details>

--- a/torchci/clickhouse_queries/advisor_verdict_for_job/query.sql
+++ b/torchci/clickhouse_queries/advisor_verdict_for_job/query.sql
@@ -4,8 +4,7 @@
 -- granules.
 SELECT
     verdict,
-    confidence,
-    run_id
+    confidence
 FROM
     misc.autorevert_advisor_verdicts
 WHERE

--- a/torchci/lib/advisor/advisorBadge.ts
+++ b/torchci/lib/advisor/advisorBadge.ts
@@ -7,6 +7,8 @@
 //   - selectAdvisorLines / renderVerdictLine / renderInProgressLine: the
 //     "AI verdict:" line emitted under each new failure in the comment.
 
+import _ from "lodash";
+
 // The signal_key prefix for HUD-originated (Dr.CI) advisor dispatches. Mirrors
 // signalKeyForJob() in advisorDispatch.ts; duplicated here to keep this module
 // free of any server-only imports.
@@ -101,21 +103,13 @@ export function verdictBadge(
   return { label: "inconclusive", color: "#8b949e", darkText: false };
 }
 
-function escapeXml(s: string): string {
-  return s
-    .replace(/&/g, "&amp;")
-    .replace(/</g, "&lt;")
-    .replace(/>/g, "&gt;")
-    .replace(/"/g, "&quot;");
-}
-
 // A minimal flat single-segment SVG pill (shields "flat" look, no left label).
 export function renderBadgeSvg(badge: AdvisorBadge): string {
   const text = badge.label;
   // Approximate text width; Verdana ~6.5px/char at 11px, plus horizontal pad.
   const width = Math.max(46, Math.round(text.length * 6.5) + 16);
   const textColor = badge.darkText ? "#33333a" : "#ffffff";
-  const safe = escapeXml(text);
+  const safe = _.escape(text);
   return `<svg xmlns="http://www.w3.org/2000/svg" width="${width}" height="20" role="img" aria-label="${safe}">
 <title>${safe}</title>
 <rect width="${width}" height="20" rx="3" fill="${badge.color}"/>
@@ -183,7 +177,7 @@ export function renderVerdictLine(
   // content, so HTML-escape it before embedding in the comment: collapse
   // newlines (can't break the blockquote) and neutralize markup so it can't
   // close the <details>/<blockquote> or inject tags.
-  const oneLine = escapeXml((summary || "").replace(/\s*\n\s*/g, " ").trim());
+  const oneLine = _.escape((summary || "").replace(/\s*\n\s*/g, " ").trim());
   return (
     `  <details><summary>AI verdict: <a href="${link}"><img src="${badge}"></a></summary><blockquote>\n\n` +
     `  ${oneLine}\n\n` +

--- a/torchci/pages/api/drci/drci.ts
+++ b/torchci/pages/api/drci/drci.ts
@@ -871,6 +871,10 @@ export function constructResultsComment(
     );
   }
 
+  // advisorLines is threaded only into the NEW FAILURES and UNCLASSIFIED
+  // sections below -- those are the only failures the advisor dispatches on.
+  // Broken-trunk / flaky / unstable / awaiting-approval jobs are already
+  // explained by their own section, so they intentionally get no verdict line.
   if (newFailedJobs.length) {
     output += constructResultsJobsSections(
       hudBaseUrl,

